### PR TITLE
Feature: doctors screen 

### PR DESCRIPTION
**Pull request by Latiah**
**What does this PR do?**

This PR includes designs of favorite doctors, top doctors and list of all doctors where you can filter and find any doctor registered in the system of the app

**Description of Task to be completed?**

Add photos of the doctors and their information including reveiws and the hospitals their work on and their roles
create needed components for the designs like the cards ,buttons

**How should this be manually tested?**

You will click on the **heart icon** on the Home screen to go to **favorite screen** and on the favorite doctor screen then click on one of the cards where you will go to a screen of removing a doctor from the favoritates .On the home screen too you will click on the **see all** which is next to top doctors and you will be redirected to the **top doctor screen** ,there you will click on the search icon where you will be directed to the search screen which will load and after sometime you will be redirected to the not found screen and  on the not found you can click on  **deafult** you will be directed to a page where doctors were found and their  list and when you click on **default**  you will be directed to the filter screen .

**Any background context you want to provide?**
N/A

**What are the relevant pivotal tracker/Trello stories?**

https://github.com/atlp-rwanda/commanders-rn-medica/issues/19

**Screenshoots of the screens designed in this pr**

Screenshot 2024-05-10 at 14 00 08

Screenshot 2024-05-10 at 14 00 08

Screenshot 2024-05-10 at 14 00 08

Screenshot 2024-05-10 at 14 00 08

Screenshot 2024-05-10 at 14 00 08

Screenshot 2024-05-10 at 14 00 08

Screenshot 2024-05-10 at 14 00 08

### CODE REVIEW

**Review by 911samuel — APPROVED** [review]

**Review by twizerim — APPROVED** [review]

File: app/Doctors/doctorsFound.tsx [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+ </View>
+ <View style={styles.btns}>
+ <TouchableOpacity style={styles.regularBn}>
+    <Text style={styles.buttonText} onPress={() => {
+                router.push("Doctors/doctorsFound")}}>Reset</Text>
+</TouchableOpacity>
+<TouchableOpacity style={styles.activeBn}>
+    <Text style={styles.activeTxt} onPress={() => {
+                router.push("Doctors/doctorsFound")}}>Apply</Text>
+</TouchableOpacity>
```
Comment (fredshema) [inline app/Doctors/doctorsFound.tsx]: These button should hide the modal by setting the `isVisible` state not routing to the same page. When clicked they go to the same page but the modal doesn't close.

File: assets/doctors/backArrow.png
Comment (fredshema) [inline assets/doctors/backArrow.png:1]: There is a back icon already in icons folder called `back.png` use that instead

File: app/(tabs)/index.tsx [hunk 2]
```diff
@@ -132,7 +134,9 @@ const Home = () => {
         />
       </View>
       <View className="flex-row items-center w-full justify-between px-6 mb-3">
-        <Text className="text-[20px] font-['Urbanist-Bold']">Top Doctors</Text>
+        <Text className="text-[20px] font-['Urbanist-Bold']"  onPress={() => {
+                router.push("/Doctors/topDoctors");
+              }}>Top Doctors</Text>
```
Comment (fredshema) [inline app/(tabs)/index.tsx]: This link to top doctors should be on the `See All` text right next to `Top Doctors`

File: app/Doctors/favoriteDoctors.tsx [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+                  <View style={styles.btns}>
+                    <TouchableOpacity style={styles.regularBtn}>
+                      <Text style={styles.btnText}  onPress={() => {
+                router.push("Doctors/favoriteDoctors")}}>Cancel</Text>
+                    </TouchableOpacity>
+                    <TouchableOpacity style={styles.activeBtn}>
+                      <Text style={styles.activeText} onPress={() => {
+                router.push("Doctors/favoriteDoctors")}}>Yes, Remove</Text>
+                    </TouchableOpacity>
+                  </View>
```
Comment (fredshema) [inline app/Doctors/favoriteDoctors.tsx]: Same here, these should set the `isVisible` state. Routing to the same page doesn't close the modals

**Review by fredshema — CHANGES_REQUESTED** [review]
There are some pages that might need little adjustments, for example:

- Searching pages on fav doctors and top doctors



The search input need a little padding on the right to not touch the end of the screen and look through the reset password screens for a rotating spinner.

- The `ScrollView` on the fav doctors and top doctors can be placed below the NavHeader



As you can see on the status bar the text go right through it when scrolling. If the scrollview is below the nav header the text will go right under the NavHeader as it can be seen on the home page.

File: app/Doctors/doctorsFound.tsx (hunk 1 above)
Comment (Latiah) [reply app/Doctors/doctorsFound.tsx]: Thank you let me work on that

File: app/(tabs)/index.tsx (hunk 2 above)
Comment (Latiah) [reply app/(tabs)/index.tsx]: Okay I was a little confused on where to link it .Let me change it

File: app/Doctors/favoriteDoctors.tsx (hunk 3 above)
Comment (Latiah) [reply app/Doctors/favoriteDoctors.tsx]: yeah I will consider that. Thank you

**Comment by Latiah** [comment]
Hey @fredshema  I worked on the changes you requested .Only one is missing which is for the ScrollView .

**Review by justinmihigo — APPROVED** [review]

**Comment by fredshema** [comment]
Can you also change the name of your pr to match the style of our team

**Review by fredshema — APPROVED** [review]

**Comment by Adadie** [comment]
@Latiah squash your commits into one and follow the branch naming conventions!

File: app/(tabs)/index.tsx [hunk 4]
```diff
@@ -137,6 +141,9 @@ const Home = () => {
         <Text className="text-[20px] font-['UrbanistBold']">Top Doctors</Text>
         <TouchableOpacity activeOpacity={0.8}>
           <Text className="text-[16px] font-['UrbanistBold'] text-primary-500">
```
Comment (Adadie) [inline app/(tabs)/index.tsx]: @Latiah Please properly close your tags and double check your codes before pushing. This bug is stopping your app from being run!!

File: app/Doctors/searchDoctor.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+};
+  useEffect(() => {
+  
+      spin();
+     
+    }
+  )
+    return (
+
+        <View className="flex-1 bg-white px-4 py-5">
```
Comment (Adadie) [inline app/Doctors/searchDoctor.tsx]: Format your codes

File: app/Doctors/searchDoctor.tsx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        height:28,
+        marginHorizontal:10,
+        marginTop:15,
+          },
+ loading:{
+    flex:1,
+    justifyContent:"center",
+    alignItems:"center",
+ }
+})
```
Comment (Adadie) [inline app/Doctors/searchDoctor.tsx]: Properly format your codes!

**Review by Adadie — CHANGES_REQUESTED** [review]

**Comment by Adadie** [comment]
@fredshema @911samuel @justinmihigo @twizerim how did you review and approve this PR when there's a bug that's preventing the app from running!?

**Comment by Adadie** [comment]
@Latiah here a couple other things you should look at

1. How do I reach the top doctors screen?
2. Your tabs aren't working. I should be able to see something different when I switch tabs.
3. The search text, border, and icons should turn to blue when I'm searching. On your app the colors don't change
4. The icons in the header are doubled
5. The card at the bottom with a prompt to action for removing isn't well formatted 

Find attached screenshot of what I'm seeing on your app
WhatsApp Image 2024-05-16 at 15 22 07
WhatsApp Image 2024-05-16 at 15 22 06
WhatsApp Image 2024-05-16 at 15 22 05
WhatsApp Image 2024-05-16 at 15 22 01

File: app/Doctors/searchDoctor.tsx (hunk 6 above)
Comment (Latiah) [reply app/Doctors/searchDoctor.tsx]: They are formated well now

File: app/(tabs)/index.tsx (hunk 4 above)
Comment (Latiah) [reply app/(tabs)/index.tsx]: Yeap I won't make a mistake of pushing without double checking again

**Pull request closed without merging**